### PR TITLE
Do type checking on notifyException function

### PR DESCRIPTION
**Pull request by wordofchristian**
Addresses https://github.com/bugsnag/bugsnag-js/issues/169#issuecomment-240792642

Also added a new test and refactored some old tests.

### CODE REVIEW

File: src/bugsnag.js [hunk 1]
```diff
@@ -78,6 +78,16 @@
     if (!exception) {
       return;
     }
+
+    if (typeof exception === "string") {
+      log(
+        "Bugsnag.notifyException() was called with a string. Expected instance of Error. " +
+        "To send a custom message instantiate a new Error or use Bugsnag.notify('<string>')." +
+        " see https://docs.bugsnag.com/platforms/browsers/#reporting-handled-exceptions"
```
Comment (GrahamCampbell) [inline src/bugsnag.js]: Very thorough. 👍

Comment (snmaynard) [inline src/bugsnag.js]: Could it call through to notify here?

Comment (wordofchristian) [inline src/bugsnag.js]: Good idea

**Comment by eanakashima** [comment]
🎇  Looks good, thanks!

**Pull request merged**